### PR TITLE
feature(#1030): Add ability to create StaticJsonRpcProvider with ConnectionInfo

### PR DESCRIPTION
**Pull request by applicature-dlitvinov**
### Description
Added ```providerConnectionInfo``` to the ```Chain``` model. It helps to create providers with authorization parameters.

### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [x] This PR passes the Circle CI checks

### CODE REVIEW

**Comment by applicature-dlitvinov** [comment]
#1030 - related to an isuue.

**Comment by applicature-dlitvinov** [comment]
@Adamj1232 @mahmud-bn @taylorjdawson please review PR.

**Review by YugasVasyl — APPROVED** [review]

**Comment by applicature-dlitvinov** [comment]
@Adamj1232 Could you please review it?

**Comment by taylorjdawson** [comment]
@applicature-dlitvinov Hey thanks for contributing! We'll take a look at this and get back to you 🙌

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ node_modules
 dist/
 package-lock.json
 .rpt2_cache
-.vscode
\ No newline at end of file
+.vscode
+/.idea
```
Comment (taylorjdawson) [inline .gitignore:7]: @applicature-dlitvinov Can you remove? thanks!

File: packages/common/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/common",
-  "version": "2.1.0",
+  "version": "2.1.1",
```
Comment (taylorjdawson) [inline packages/common/package.json:3]:
```suggestion
  "version": "2.1.1-alpha.1",
```

**Review by taylorjdawson — CHANGES_REQUESTED** [review]
Just a couple comments! Also could you provide an example of how you would use this? Thanks!

**Comment by applicature-dlitvinov** [comment]
@taylorjdawson 
Sometimes we need to use a private network RPC URL with some credentials, It's more secure and more stable than native RPC (Sometimes has crashes or lags).

```
const onboard = Onboard({
      wallets: [injected],
      chains: [
        {
          id: '0x61',
          token: 'BNB',
          label: 'Binance Testnet',
          rpcUrl: `${CUCTOM_BSC_PRC}`,
          providerConnectionInfo: {
            url: `${CUCTOM_BSC_PRC}`,
            user: `${LOGIN}`,
            password: `${PASSWORD}`
          }
        },
        ...
      ]
    })
```

**Comment by applicature-dlitvinov** [comment]
@taylorjdawson Coul. you please review again

**Review by taylorjdawson — APPROVED** [review]

**Pull request merged**